// babel-core/lib/babel/transformation/transformers/internal/explode.js
"use strict";

var _interopRequireWildcard = function (obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (typeof obj === "object" && obj !== null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj["default"] = obj; return newObj; } };

var _interopRequireDefault = function (obj) { return obj && obj.__esModule ? obj : { "default": obj }; };

exports.__esModule = true;

var _clone = require("lodash/lang/clone");

var _clone2 = _interopRequireDefault(_clone);

var _import = require("../../../types");

var t = _interopRequireWildcard(_import);

var metadata = {
  group: "builtin-setup"
};

exports.metadata = metadata;
function buildClone(bindingKey, refKey) {
  return function (node) {
    if (node[bindingKey] === node[refKey]) {
      node[refKey] = t.removeComments(_clone2["default"](node[refKey]));
    }
  };
}

var Property = buildClone("value", "key");
exports.Property = Property;
var ExportSpecifier = buildClone("local", "exported");
exports.ExportSpecifier = ExportSpecifier;
var ImportSpecifier = buildClone("local", "imported");
exports.ImportSpecifier = ImportSpecifier;
